From 8274c99b34b0302a7f9c7a2bf33e38e4e61f82b1 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Wed, 3 Jan 2018 21:41:24 +0900
Subject: [PATCH] Make priority change work

To get pid number from GTK selection, column id must be set to COLUMN_PID.
Also, to redirect g_spawn_command_line_sync() result to /dev/null using
shell-style redirect, shell command must be used, otherwise such redirection
is regarded as command's arguments.
---
 src/callbacks.c              | 2 +-
 src/xfce-taskmanager-linux.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/callbacks.c b/src/callbacks.c
index dfaff57..315852e 100644
--- a/src/callbacks.c
+++ b/src/callbacks.c
@@ -103,7 +103,7 @@ void handle_prio_menu(GtkWidget *widget, gchar *prio)
 
     if(gtk_tree_selection_get_selected(selection, &model, &iter))
     {
-        gtk_tree_model_get(model, &iter, 1, &task_id, -1);
+        gtk_tree_model_get(model, &iter, COLUMN_PID, &task_id, -1);
 
         set_priority_to_task(atoi(task_id), atoi(prio));
         refresh_task_list();
diff --git a/src/xfce-taskmanager-linux.c b/src/xfce-taskmanager-linux.c
index 36656fb..dd490c1 100644
--- a/src/xfce-taskmanager-linux.c
+++ b/src/xfce-taskmanager-linux.c
@@ -306,7 +306,7 @@ void set_priority_to_task(gint task_id, gint prio)
     {
     	int status = 0;
         gchar command[128] = "";
-        g_sprintf(command, "renice %d %d > /dev/null", prio, task_id);
+        g_sprintf(command, "sh -c \"renice %d %d > /dev/null\"", prio, task_id);
 
         if( ! g_spawn_command_line_sync(command, NULL, NULL, &status, NULL) ||status )
             show_error(_("Couldn't set priority %d to the task with ID %d"), prio, task_id);
-- 
2.1.4

